client/state/notices/middleware.js

```diff
@@ -122,6 +122,12 @@ export const onPublicizeConnectionCreate = ( dispatch, { connection } ) => dispa
 	} ) )
 );
 
+export const onPublicizeConnectionCreateFailure = ( dispatch, { error } ) => dispatch(
+	errorNotice( error.message || translate( 'An error occurred while connecting the account.', {
```

Comment:
You can also test a bit more quickly in dev via: `dispatch( { type: 'PUBLICIZE_CONNECTION_CREATE_FAILURE', error: { message: 'foo' } } )`